event_profiling.h

```diff
@@ -0,0 +1,175 @@
+#ifndef USE_EVENT_PROFILING
+#define USE_EVENT_PROFILING 0
+#endif
+
+#ifndef RUBY_EVENT_PROFILING_H
+#define RUBY_EVENT_PROFILING_H 1
+
+#if USE_EVENT_PROFILING
+
+#include "ruby/internal/config.h"
+
+#include "vm_core.h"
+
+#include <pthread.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+
+#if !defined(__GNUC__) && USE_EVENT_PROFILING
+#error "USE_EVENT_PROFILING is not supported by other than __GNUC__"
+#endif
+
+/* To avoid using realloc, allocate large heap memory ahead of execution. */
+typedef struct rb_event_profiling_config
+{
+    int max_ractors;
+    int max_ractor_events;
+    int max_call_stack_depth;
+} rb_event_profiling_config_t;
+
+typedef enum rb_profiling_event_phase
```

Comment:
The file (component) name is `event_profiling`, so the prefix can be `rb_event_profiling_`.

`rb_profiling_event_phase_t` -> `rb_event_profiling_phase_t`.

BTW, `rb_profiling_event_phase_t` is used only once in this patch, so you don't need to make new type.